### PR TITLE
lambda AnsibleAWSModule cleanup (also enable retries)

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Various AnsibleAWSModule cleanup

- Use module.client helpers
- Use is_boto3_error_code('AccessDenied')
- Delete dead code paths
- Use AWSRetries (now that they don't automatically include NotFound exceptions)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lambda

### CODE REVIEW

File: plugins/modules/lambda.py [hunk 1]
```diff
@@ -241,27 +244,22 @@ def get_account_info(module, region=None, endpoint=None, **aws_connect_kwargs):
...
-            if (e.response['Error']['Code'] == 'AccessDenied'):
-                except_msg = to_native(e.message)
-                m = except_msg.search(r"arn:(aws(-([a-z\-]+))?):iam::([0-9]{12,32}):\w+/")
-                account_id = m.group(4)
-                partition = m.group(1)
-            if account_id is None:
+            iam_client = module.client('iam', retry_decorator=AWSRetry.jittered_backoff())
+            arn, partition, service, reg, account_id, resource = iam_client.get_user(aws_retry=True)['User']['Arn'].split(':')
+        except is_boto3_error_code('AccessDenied') as e:
+            except_msg = to_native(e.message)
```
Comment (s-hertel) [inline plugins/modules/lambda.py:256]: e.message doesn't exist for Python 3. `fail_json_aws` does a try/except like
```python
        try:
            except_msg = to_native(exception.message)
        except AttributeError:
            except_msg = to_native(exception)
```

Comment (tremble) [reply plugins/modules/lambda.py]: Updated

File: plugins/modules/lambda.py [hunk 2]
```diff
@@ -237,27 +240,25 @@ def get_account_info(module, region=None, endpoint=None, **aws_connect_kwargs):
...
-                except_msg = to_native(e.message)
-                m = except_msg.search(r"arn:(aws(-([a-z\-]+))?):iam::([0-9]{12,32}):\w+/")
-                account_id = m.group(4)
-                partition = m.group(1)
-            if account_id is None:
+            iam_client = module.client('iam', retry_decorator=AWSRetry.jittered_backoff())
+            arn, partition, service, reg, account_id, resource = iam_client.get_user(aws_retry=True)['User']['Arn'].split(':')
+        except is_boto3_error_code('AccessDenied') as e:
+            try:
+                except_msg = to_native(exception.message)
```
Comment (s-hertel) [inline plugins/modules/lambda.py:253]: Sorry, I copy-pasted that piece of code from module_utils
```suggestion
                except_msg = to_native(e.message)
```

File: plugins/modules/lambda.py [hunk 3]
```diff
@@ -237,27 +240,25 @@ def get_account_info(module, region=None, endpoint=None, **aws_connect_kwargs):
...
+            except AttributeError:
+                except_msg = to_native(exception)
```
Comment (s-hertel) [inline plugins/modules/lambda.py:255]:
```suggestion
                except_msg = to_native(e)
```

**Comment by tremble** [comment]
Yay green tests!  Thanks for the fix in #167 @jillr

**Comment by tremble** [comment]
@jillr based on your comment I'm going to merge this, only change was a simple rebase and the tests are now green!

> jillr: tremble: and that fix left #5 with another merge conflict  :)  once everything is green you can merge that one

**Pull request merged**